mailpoet/lib/Segments/SubscribersFinder.php

```diff
@@ -42,6 +43,18 @@ public function findSubscribersInSegments($subscribersToProcessIds, $newsletterS
       }
       $result = array_merge($result, $this->findSubscribersInSegment($segment, $subscribersToProcessIds));
     }
+
+    if (is_int($filterSegmentId)) {
+      $filterSegment = $this->segmentsRepository->findOneById($filterSegmentId);
+      if ($filterSegment instanceof SegmentEntity) {
+        if ($filterSegment->getType() !== SegmentEntity::TYPE_DYNAMIC) {
+          throw new UnexpectedValueException("Filter segment ID must be a dynamic segment. Type of filter with ID {$filterSegment->getId()} is {$filterSegment->getType()}.");
```

Comment:
Thanks for pointing this out! It definitely seems like something we shouldn't allow. I'll try to come up with a better solution.

Reply:
I addressed this issue in 39c1ac1c449cfce3e1372828da0ba64495c738f4. It's still possible to end up in a kind of strange state when a newsletter has a filter segment that belongs to a static list, where it completes but sends to nobody. The same state seems possible without this change by scheduling an email to send to a list and then removing all subscribers from that list. Since this very much an edge case and we'll do our best to prevent saving a list ID as a filter segment ID, I don't know that this warrants a more elegant solution.

Please let me know if you can think of a better way of handling this, or if you're still able to end up in a broken sending state.

Reply:
The approach that you implemented looks good to me and I agree that there is no need to cover this other edge case that you described.